p42/src/song.rs

```diff
@@ -0,0 +1,176 @@
_string(iter_with_line_num()), "1: On the first day of Christmas,\n2: My true love gave to me\n3: A partridge in a pear tree.\n4: \n5: On the second day of Christmas,\n6: My true love gave to me\n7: Two turtle doves,\n8: And a partridge in a pear tree.\n9: \n10: On the third day of Christmas,\n11: My true love gave to me\n12: Three French hens,\n13: Two turtle doves,\n14: And a partridge in a pear tree.\n15: \n16: On the fourth day of Christmas,\n17: My true love gave to me\n18: Four calling birds,\n19: Three French hens,\n20: Two turtle doves,\n21: And a partridge in a pear tree.\n22: \n23: On the fifth day of Christmas,\n24: My true love gave to me\n25: Five golden rings,\n26: Four calling birds,\n27: Three French hens,\n28: Two turtle doves,\n29: And a partridge in a pear tree.\n30: \n31: On the sixth day of Christmas,\n32: My true love gave to me\n33: Six geese a-laying,\n34: Five golden rings,\n35: Four calling birds,\n36: Three French hens,\n37: Two turtle doves,\n38: And a partridge in a pear tree.\n39: \n40: On the seventh day of Christmas,\n41: My true love gave to me\n42: Seven swans a-swimming,\n43: Six geese a-laying,\n44: Five golden rings,\n45: Four calling birds,\n46: Three French hens,\n47: Two turtle doves,\n48: And a partridge in a pear tree.\n49: \n50: On the eighth day of Christmas,\n51: My true love gave to me\n52: Eight maids a-milking,\n53: Seven swans a-swimming,\n54: Six geese a-laying,\n55: Five golden rings,\n56: Four calling birds,\n57: Three French hens,\n58: Two turtle doves,\n59: And a partridge in a pear tree.\n60: \n61: On the ninth day of Christmas,\n62: My true love gave to me\n63: Nine ladies dancing,\n64: Eight maids a-milking,\n65: Seven swans a-swimming,\n66: Six geese a-laying,\n67: Five golden rings,\n68: Four calling birds,\n69: Three French hens,\n70: Two turtle doves,\n71: And a partridge in a pear tree.\n72: \n73: On the tenth day of Christmas,\n74: My true love gave to me\n75: Ten lords a-leaping,\n76: Nine ladies dancing,\n77: Eight maids a-milking,\n78: Seven swans a-swimming,\n79: Six geese a-laying,\n80: Five golden rings,\n81: Four calling birds,\n82: Three French hens,\n83: Two turtle doves,\n84: And a partridge in a pear tree.\n85: \n86: On the eleventh day of Christmas,\n87: My true love gave to me\n88: Eleven pipers piping,\n89: Ten lords a-leaping,\n90: Nine ladies dancing,\n91: Eight maids a-milking,\n92: Seven swans a-swimming,\n93: Six geese a-laying,\n94: Five golden rings,\n95: Four calling birds,\n96: Three French hens,\n97: Two turtle doves,\n98: And a partridge in a pear tree.\n99: \n100: On the twelfth day of Christmas,\n101: My true love gave to me\n102: Twelve drummers drumming,\n103: Eleven pipers piping,\n104: Ten lords a-leaping,\n105: Nine ladies dancing,\n106: Eight maids a-milking,\n107: Seven swans a-swimming,\n108: Six geese a-laying,\n109: Five golden rings,\n110: Four calling birds,\n111: Three French hens,\n112: Two turtle doves,\n113: And a partridge in a pear tree.");
```

Comment:
It's better to make doctests relatively small, since they are part of crate documentation. It's better to keep such tests as unit or integration tests. You also can use [`include_str!`](https://doc.rust-lang.org/std/macro.include_str.html) to keep the lyrics in a separate file.